File: f1-api/controllers/mock/calendar.go
package mock

import "f1-api/model"

// CalendarController :
type CalendarController struct{}

// GetCalendar :
func (cc CalendarController) GetCalendar() (model.Calendar, error) {

	calendar := model.Calendar{
		Season: "Season1",
		Events: []model.Event{
			model.Event{
				Day:     "Day1",
				Circuit: "Circuit1",
			},
			model.Event{
				Day:     "Day2",
				Circuit: "Circuit2",
			},
			model.Event{
				Day:     "Day3",
				Circuit: "Circuit3",
			},
		},
	}

	return calendar, nil
}

File: f1-discord-bot/model/calendar.js
class Calendar{

    constructor(season = '', events = [] ){

        this.season = season
        this.events = events
    }

    static fromJSON(jsonCalendar){

        let season = ''
        let events = []

        if (jsonCalendar.hasOwnProperty('Season')){
            season = jsonCalendar['Season']
        }

        if (jsonCalendar.hasOwnProperty('Events')){

            let jsonEvents = jsonCalendar['Events']

            jsonEvents.forEach(jsonEvent => {
                events.push(Event.fromJSON(jsonEvent))
            })
        }

        return new Calendar(season = season, events = events)
    }
}


class Event{

    constructor(day = '', circuit = ''){
        
        this.day = day
        this.circuit = circuit
    }

    static fromJSON(jsonEvent){

        let day = ''
        let circuit = ''

        if (jsonEvent.hasOwnProperty('Day')){
            day = jsonEvent['Day']
        }

        if (jsonEvent.hasOwnProperty('Circuit')){
            circuit = jsonEvent['Circuit']
        }

        return new Event(day = day, circuit = circuit)
    }
}

module.exports = {
    Calendar: Calendar,
    Event: Event
}